ocean_lib/ocean/ocean_assets.py

```diff
@@ -322,48 +375,185 @@ def create(
             logger.error(msg)
             raise ValueError(msg)
 
-        # Process the DDO
-        asset_dict = asset.as_dictionary()
-        ddo_string = json.dumps(asset_dict, separators=(",", ":"))
-        ddo_bytes = ddo_string.encode("utf-8")
-        ddo_hash = create_checksum(ddo_string)
+        document, flags, ddo_hash = self.process_ddo(
+            asset, provider_uri, encrypt_flag, compress_flag
+        )
 
-        # Plain asset
-        if not encrypt_flag and not compress_flag:
-            flags = bytes([0])
-            document = ddo_bytes
+        _ = erc721_token.set_metadata(
+            metadata_state=0,
+            metadata_decryptor_url=provider_uri,
+            metadata_decryptor_address=publisher_wallet.address,
+            flags=flags,
+            data=document,
+            data_hash=ddo_hash,
+            data_proofs=[],
+            from_wallet=publisher_wallet,
+        )
 
-        # Only compression, not encrypted
-        elif compress_flag and not encrypt_flag:
-            flags = bytes([1])
-            # Compress DDO
-            document = lzma.compress(ddo_bytes)
+        # Fetch the asset on chain
+        asset = self._get_aquarius(self._metadata_cache_uri).wait_for_asset(did)
 
-        # Only encryption, not compressed
-        elif encrypt_flag and not compress_flag:
-            flags = bytes([2])
-            # Encrypt DDO
-            encrypt_response = DataServiceProvider.encrypt(
-                objects_to_encrypt=ddo_string,
-                encrypt_endpoint=f"{provider_uri}/api/services/encrypt",
+        return asset
+
+    def update(
+        self,
+        did: str,
+        new_metadata: dict,
+        publisher_wallet: Wallet,
+        delete_erc20_tokens: Optional[List[str]] = [],
```

Comment:
can't you figure this parameter out by comparing the current list against the new list?

Reply:
Yes, I guess it makes sense, I wanted to be sure the deletion of an datatoken from the NFT was aware. With the change you  proposed of passing the modified Asset it makes sense to me to get rid of this.
Thank you.